src/main/java/jp/co/future/uroborosql/SqlQueryImpl.java

```diff
@@ -43,123 +44,142 @@
+	 * @see jp.co.future.uroborosql.fluent.SqlQuery#findFirst(java.lang.Class)
 	 */
 	@Override
-	public Map<String, Object> first() {
-		return first(agent().getDefaultMapKeyCaseFormat());
+	public <T> Optional<T> findFirst(final Class<T> type) {
+		try (Stream<T> stream = stream(new EntityResultSetConverter<>(type, new PropertyMapperManager()))) {
+			return stream.findFirst();
+		}
 	}
 
 	/**
 	 * {@inheritDoc}
 	 *
-	 * @see jp.co.future.uroborosql.fluent.SqlQuery#first(jp.co.future.uroborosql.utils.CaseFormat)
+	 * @see jp.co.future.uroborosql.fluent.SqlQuery#one()
 	 */
 	@Override
-	public Map<String, Object> first(final CaseFormat caseFormat) {
-		return findFirst(caseFormat).orElseThrow(() -> new DataNotFoundException("query result is empty."));
+	public Map<String, Object> one() {
+		return one(agent().getDefaultMapKeyCaseFormat());
 	}
 
 	/**
 	 * {@inheritDoc}
 	 *
-	 * @see jp.co.future.uroborosql.fluent.SqlQuery#first(Class)
+	 * @see jp.co.future.uroborosql.fluent.SqlQuery#one(CaseFormat)
 	 */
 	@Override
-	public <T> T first(final Class<T> type) {
-		return findFirst(type).orElseThrow(() -> new DataNotFoundException("query result is empty."));
+	public Map<String, Object> one(final CaseFormat caseFormat) {
+		return findOne(caseFormat).orElseThrow(() -> new DataNotFoundException("query result is empty."));
 	}
 
 	/**
 	 * {@inheritDoc}
 	 *
-	 * @see jp.co.future.uroborosql.fluent.SqlQuery#findFirst()
+	 * @see jp.co.future.uroborosql.fluent.SqlQuery#one(Class)
 	 */
 	@Override
-	public Optional<Map<String, Object>> findFirst() {
-		return findFirst(agent().getDefaultMapKeyCaseFormat());
+	public <T> T one(final Class<T> type) {
+		return findOne(type).orElseThrow(() -> new DataNotFoundException("query result is empty."));
 	}
 
 	/**
 	 * {@inheritDoc}
 	 *
-	 * @see jp.co.future.uroborosql.fluent.SqlQuery#findFirst(jp.co.future.uroborosql.utils.CaseFormat)
+	 * @see jp.co.future.uroborosql.fluent.SqlQuery#findOne()
 	 */
 	@Override
-	public Optional<Map<String, Object>> findFirst(final CaseFormat caseFormat) {
+	public Optional<Map<String, Object>> findOne() {
+		return findOne(agent().getDefaultMapKeyCaseFormat());
+	}
+
+	@Override
+	public Optional<Map<String, Object>> findOne(final CaseFormat caseFormat) {
 		try (Stream<Map<String, Object>> stream = stream(
-				new MapResultSetConverter(agent.getSqlConfig().getDialect(), caseFormat))) {
-			return stream.findFirst();
+				new MapResultSetConverter(this.agent.getSqlConfig().getDialect(), caseFormat))) {
+			List<Map<String, Object>> resultList = stream.limit(2).collect(Collectors.toList());
+			if (resultList.size() > 1) {
+				throw new DataNonUniqueException("two or more query results.");
+			}
+			return resultList.stream().findFirst();
+		}
+	}
+
+	@Override
+	public <T> Optional<T> findOne(final Class<T> type) {
+		try (Stream<T> stream = stream(new EntityResultSetConverter<>(type, new PropertyMapperManager()))) {
```

Comment:
use `stream(type)` instead of `stream(new EntityResultSetConverter<>(type, new PropertyMapperManager()))`